cql3/user_types.cc

```diff
@@ -229,4 +229,58 @@ cql3::raw_value_view user_types::delayed_value::bind_and_get(const query_options
     return options.make_temporary(cql3::raw_value::make_value(user_type_impl::build_value(bind_internal(options))));
 }
 
+void user_types::setter::execute(mutation& m, const clustering_key_prefix& row_key, const update_parameters& params) {
+    auto value = _t->bind(params._options);
+    if (value == constants::UNSET_VALUE) {
+        return;
+    }
+
+    auto& type = static_cast<const user_type_impl&>(*column.type);
+    if (type.is_multi_cell()) {
+        // Non-frozen user defined type.
+
+        collection_mutation_description mut;
+
+        // Setting a non-frozen (multi-cell) UDT means overwriting all cells.
+        // We start by deleting all existing cells.
+
+        // TODO? (kbraun): is this the desired behaviour?
```

Comment:
Yes, collections work the same way. I guess this nullifies my unset comment.

Reply:
For collections, you can use SET c = c + ? to merge into the current value (and not create a tombstone).